### PR TITLE
Fixed compiler issues with Swift 5.8 and macCatalyst.

### PR DESCRIPTION
**Pull request by kudit**
Removed xrOS references, and added checks that enable this package to be included in macCatalyst apps without compiler errors.  Also fixes so that there are no compiler errors when using Swift 5.8 or earlier.  Added an `orientation` property if on iOS that wraps the UIDevice current orientation if UIDevice is available.

### CODE REVIEW

**Comment by Zandor300** [comment]
@kudit I'm sorry but your PR contains a lot of different changes. I can't accept your PR unless split it up into separate PRs with smaller changes. I closed your other PR for the same reason:
- https://github.com/devicekit/DeviceKit/pull/369

**Comment by kudit** [comment]
They're all related.  It's all checks to fix compiler issues when building for macOS (which doesn't support UIKit) and Swift 5.8 (which doesn't recognize xrOS or visionOS `os(` tags).  Only thing that might be able to be separated is the orientation code but other than that, all those checks are needed to remove compiler warnings.  It's a lot of code because there are so many places where UIKit is assumed and the alternate code path wasn't provided/implemented.  If you review, you'll see that it's all pretty simple and straight forward.   Is there a way to not include the generated file so that the code only needs to be updated in one place rather than two nearly identical files?

File: Source/Device.swift.gyb [hunk 1]
```diff
@@ -303,6 +303,7 @@ import WatchKit
 import UIKit
 #else
 import Foundation
+import IOKit
```
Comment (Zandor300) [inline Source/Device.swift.gyb:306]: Please move macOS related code to a separate PR

File: Source/Device.swift.gyb [hunk 2]
```diff
@@ -418,6 +419,11 @@ public enum Device {
       case "i386", "x86_64", "arm64": return simulator(mapToDevice(identifier: ProcessInfo().environment["SIMULATOR_MODEL_IDENTIFIER"] ?? "watchOS"))
       default: return unknown(identifier)
       }
+#elseif os(macOS)
+  switch identifier {
+  case "Mac14,6": return macBookProM2Max
+  default: return unknown(identifier)
+  }
```
Comment (Zandor300) [inline Source/Device.swift.gyb:426]: Please move macOS-related code to a separate PR. Also, please use the gyb-mechanism to generate this code instead of hardcoding it.

File: Source/Device.swift.gyb [hunk 3]
```diff
@@ -708,6 +714,10 @@ public enum Device {
     public var hasForceTouchSupport: Bool {
       return isOneOf(Device.allWatchesWithForceTouchSupport) || isOneOf(Device.allWatchesWithForceTouchSupport.map(Device.simulator))
     }
+  #elseif os(macOS)
+  public static var allMacs: [Device] {
+    return [.macBookProM2Max] // TODO: add others
+  }
```
Comment (Zandor300) [inline Source/Device.swift.gyb:720]: Please move macOS related code to a separate PR. Also needs to be generated using gyb.

File: Source/Device.swift.gyb [hunk 4]
```diff
@@ -728,6 +738,8 @@ public enum Device {
       return allTVs
     #elseif os(watchOS)
       return allWatches
+    #elseif os(macOS)
+      return allMacs
```
Comment (Zandor300) [inline Source/Device.swift.gyb:742]: macOS-related code.

File: Source/Device.swift.gyb [hunk 5]
```diff
@@ -789,8 +801,17 @@ public enum Device {
...
     return UIDevice.current.name
+    #elseif canImport(Foundation)
+    return ProcessInfo.processInfo.hostName
+    #else
+    // pulled from https://github.com/rjstelling/Hostess.swift/blob/master/Projects/Hostess/Hostess/Hostess.swift
+    var hostname = [CChar](repeating: 0x0, count: Int(NI_MAXHOST))
+    guard gethostname(&hostname, Int(NI_MAXHOST)) == noErr else {
+        return nil
+    }
+    return String(cString: hostname)
```
Comment (Zandor300) [inline Source/Device.swift.gyb:814]: macOS-related code.

File: Source/Device.swift.gyb [hunk 6]
```diff
@@ -805,8 +826,12 @@ public enum Device {
     } else {
       return UIDevice.current.systemName
     }
-    #else
+    #elseif canImport(UIKit)
     return UIDevice.current.systemName
+    #elseif os(macOS)
+      return "macOS"
```
Comment (Zandor300) [inline Source/Device.swift.gyb:832]: macOS-related code.

File: Source/Device.swift.gyb [hunk 7]
```diff
@@ -815,8 +840,10 @@ public enum Device {
     guard isCurrent else { return nil }
     #if os(watchOS)
     return WKInterfaceDevice.current().systemVersion
-    #else
+  #elseif canImport(UIKit)
     return UIDevice.current.systemVersion
+    #else
+    return ProcessInfo.processInfo.operatingSystemVersionString
```
Comment (Zandor300) [inline Source/Device.swift.gyb:846]: macOS-related code.

File: Source/Device.swift.gyb [hunk 8]
```diff
@@ -835,8 +864,10 @@ public enum Device {
     guard isCurrent else { return nil }
     #if os(watchOS)
     return WKInterfaceDevice.current().localizedModel
-    #else
+    #elseif canImport(UIKit)
     return UIDevice.current.localizedModel
+    #else
+    return Host.current().localizedName
```
Comment (Zandor300) [inline Source/Device.swift.gyb:870]: macOS-related code.

File: Source/Device.swift.gyb [hunk 9]
```diff
@@ -825,8 +852,10 @@ public enum Device {
     guard isCurrent else { return nil }
     #if os(watchOS)
     return WKInterfaceDevice.current().model
-    #else
+#elseif canImport(UIKit)
     return UIDevice.current.model
+    #else
+    return Host.current().name
```
Comment (Zandor300) [inline Source/Device.swift.gyb:858]: macOS-related code.

File: Source/Device.swift.gyb [hunk 10]
```diff
@@ -915,6 +946,12 @@ extension Device: CustomStringConvertible {
       case .simulator(let model): return "Simulator (\(model.description))"
       case .unknown(let identifier): return identifier
       }
+    #else
+    switch self {
+    case .macBookProM2Max: return "MacBook Pro M2 Max 16\""
+    case .simulator(let model): return "Simulator (\(model.description))"
+    case .unknown(let identifier): return identifier
+    }
```
Comment (Zandor300) [inline Source/Device.swift.gyb:954]: macOS-related code.

File: Source/Device.swift.gyb [hunk 11]
```diff
@@ -947,6 +984,12 @@ extension Device: CustomStringConvertible {
       case .simulator(let model): return "Simulator (\(model.safeDescription))"
       case .unknown(let identifier): return identifier
       }
+    #else
+      switch self {
+      case .macBookProM2Max: return "MacBook Pro M2 Max 16in"
+      case .simulator(let model): return "Simulator (\(model.description))"
+      case .unknown(let identifier): return identifier
+      }
```
Comment (Zandor300) [inline Source/Device.swift.gyb:992]: macOS-related code.

File: Source/Device.swift.gyb [hunk 12]
```diff
@@ -1405,6 +1474,14 @@ extension Device {
...
     case ${cpu.name}
 % end
+  #elseif os(macOS)
+    case i3
+    case i5
+    case i7
+    case i9
+    case m1
+    case m2
+    case m3
```
Comment (Zandor300) [inline Source/Device.swift.gyb:1484]: macOS-related code.

File: Source/Device.swift.gyb [hunk 13]
```diff
@@ -1435,6 +1512,12 @@ extension Device {
       case .simulator(let model): return model.cpu
       case .unknown: return .unknown
     }
+#else
+switch self {
+case .macBookProM2Max: return .m2
+case .simulator(let model): return model.cpu
+case .unknown: return .unknown
+}
```
Comment (Zandor300) [inline Source/Device.swift.gyb:1520]: macOS-related code. Also please fix the indentation.

File: Source/Device.swift.gyb [hunk 14]
```diff
@@ -1457,6 +1540,17 @@ extension Device.CPU: CustomStringConvertible {
...
+    switch self {
+      case .i3: return "Core i3"
+      case .i5: return "Core i5"
+      case .i7: return "Core i7"
+      case .i9: return "Core i9"
+      case .m1: return "M1"
+      case .m2: return "M2"
+      case .m3: return "M3"
+      case .unknown: return "unknown"
+    }
```
Comment (Zandor300) [inline Source/Device.swift.gyb:1553]: macOS-related code.

**Review by Zandor300 — CHANGES_REQUESTED** [review]
